core/integration/module_config_test.go

```diff
@@ -1166,134 +1178,211 @@ func (m *Test) Fn() ([]string, error) {
 	require.Contains(t, strings.TrimSpace(out), "random-file")
 }
 
-const (
-	gitTestRepoURL    = "github.com/dagger/dagger-test-modules"
-	gitTestRepoCommit = "2cb6cb4b0dba52c1e65b3ff46dd1a4a8f9a02f94"
-)
+// VCS to test behavior against
+type vcsTestCase struct {
+	name              string
+	gitTestRepoRef    string
+	gitTestRepoCommit string
+	baseHTMLURL       string // expected base HTML URLs, might differs from ref (e.g. not contain .git)
+	urlPathComponent  string // path separator to access `tree` view of src at commit, per provider
+	host              string // host component of repoURL
+}
+
+var vcsTestCases = []vcsTestCase{
```

Comment:
I like this helper, but why is it neccessary to test against with and without `.git`? Effectively, this x6 the tests we need to run, x3 already seems like quite a few.

Reply:
It is not necessary per se. But I wanted to have at least a few runs to make sure that all the tests were passing even with it.

This `.git` is going to be tested with private git repos, so I removed them from the matrix 🙏 x3 is already a lot I agree